test/plugin-loader/plugin-loader.js

```diff
@@ -0,0 +1,247 @@
+          {
+            "type": "router",
+            "name": "foo",
+            "fileName": "nop-router.js",
+            "version": "4.5.6"
+          },
+          {
+            "type": "router",
+            "name": "bar",
+            "fileName": "nop-router.js",
+            "version": "4.4.4"
+          }
+        ]
+      };
+      const p = makePlugin(testPluginDef, {}, pluginContext, false);
+      assert(p.dataServicesGrouped)
+      assert(p.dataServicesGrouped['foo'])
+      assert.equal(p.dataServicesGrouped['foo'].highestVersion, '4.5.6');
+      assert(p.dataServicesGrouped['foo'].versions['1.2.3'])
+      assert.equal(p.dataServicesGrouped['foo'].versions['1.2.3'].name, 'foo')
+      assert(p.dataServicesGrouped['foo'].versions['4.5.6'])
+      assert.equal(p.dataServicesGrouped['foo'].versions['4.5.6'].name, 'foo')
+      assert(p.dataServicesGrouped['bar'])
+      assert.equal(p.dataServicesGrouped['bar'].highestVersion, '4.4.4');
+      assert(p.dataServicesGrouped['bar'].versions['4.4.4'])
+      assert.equal(p.dataServicesGrouped['bar'].versions['4.4.4'].name, 'bar')
+    }),
+    
+    it('should correctly group imports', function() {
+      const testPluginDef = {
+        "identifier": "com.rs.testplugin",
+        "baseURI": "testplugin",
+        "apiVersion": "1.0.0",
+        "pluginVersion": "1.0.0",
+        "pluginType": "application",
+        "dataServices": [
+          {
+            "type": "import",
+            "sourcePlugin": "com.rs.provider",
+            "sourceName": "blah",
+            "versionRange": "2.0.0",
+            "version": "2.0.0",
+            "localName": "blah"
+          },
+          {
+            "type": "import",
+            "sourcePlugin": "com.rs.intermediary",
+            "sourceName": "something",
+            "versionRange": "1.0.0",
+            "version": "1.0.0",
+            "localName": "something"
+          }
+        ]
+      };
+      const p = makePlugin(testPluginDef, {}, pluginContext, false);
+      assert(p.importsGrouped['blah'])
+      assert.equal(p.importsGrouped['blah'].highestVersion, '2.0.0');
+      assert(p.importsGrouped['blah'].versions['2.0.0'])
+      assert(p.importsGrouped['something'])
+      assert.equal(p.importsGrouped['something'].highestVersion, '1.0.0');
+      assert(p.importsGrouped['something'].versions['1.0.0'])
+    }),
+    
+    it('should correctly check local service version dependencies', function() {
+      const twoVersionsOfAService = {
+        "identifier": "com.rs.testplugin",
+        "baseURI": "testplugin",
+        "apiVersion": "1.0.0",
+        "pluginVersion": "1.0.0",
+        "pluginType": "application",
+        "dataServices": [
+          {
+            "type": "router",
+            "name": "foo",
+            "fileName": "nop-router.js",
+            "version": "1.2.3"
+          },
+          {
+            "type": "router",
+            "name": "foo",
+            "fileName": "nop-router.js",
```

Comment:
I don't think you can actually access foo@1.2.3 and foo@4.5.6 in this situation. If you have 2 routers which come from the same file, wouldn't you need to specify individual & different factory function names with `routerFactory`? Otherwise the returned object is identical.

Reply:
Yes, but that service is never invoked, so that's irrelevant for this test's purposes